driver-core/src/main/com/mongodb/client/model/search/SearchOperator.java

```diff
@@ -292,6 +300,121 @@ static GeoNearSearchOperator near(final Point origin, final Number pivot, final
+     * Returns a {@link SearchOperator} that searches for an array of values at the given path and returns documents where the value of
+     * the field equals any value in the specified array.
+     *
+     * @param path The indexed field to be searched.
+     * @param value The objectId value to search for.
+     * @param values More fields to be searched.
+     * @return The requested {@link SearchOperator}.
+     * @mongodb.atlas.manual atlas-search/in/ in operator
+     */
+    static InSearchOperator in(final FieldSearchPath path, final ObjectId value, final ObjectId... values) {
+        return in(notNull("path", path), concat(notNull("value", value), values));
+    }
+
+    /**
+     * Returns a {@link SearchOperator} that searches for an array of values at the given path and returns documents where the value of
+     * the field equals any value in the specified array.
+     *
+     * @param path The indexed field to be searched.
+     * @param value The number value to search for.
+     * @param values More fields to be searched.
+     * @return The requested {@link SearchOperator}.
+     * @mongodb.atlas.manual atlas-search/in/ in operator
+     */
+    static InSearchOperator in(final FieldSearchPath path, final Number value, final Number... values) {
+        return in(notNull("path", path), concat(notNull("value", value), values));
+    }
+
+    /**
+     * Returns a {@link SearchOperator} that searches for an array of values at the given path and returns documents where the value of
+     * the field equals any value in the specified array.
+     *
+     * @param path The indexed field to be searched.
+     * @param value The instant date value to search for.
+     * @param values More fields to be searched.
+     * @return The requested {@link SearchOperator}.
+     * @mongodb.atlas.manual atlas-search/in/ in operator
+     */
+    static InSearchOperator in(final FieldSearchPath path, final Instant value, final Instant... values) {
+        return in(notNull("path", path), concat(notNull("value", value), values));
+    }
+
+    /**
+     * Returns a {@link SearchOperator} that searches for an array of values at the given path and returns documents where the value of
+     * the field equals any value in the specified array.
+     *
+     * @param path The indexed field to be searched.
+     * @param value The uuid value to search for.
+     * @param values More fields to be searched.
+     * @return The requested {@link SearchOperator}.
+     * @mongodb.atlas.manual atlas-search/in/ in operator
+     */
+    static InSearchOperator in(final FieldSearchPath path, final UUID value, final UUID... values) {
+        List<BsonBinary> bsonValues = new ArrayList<>();
+        bsonValues.add(new BsonBinary(value));
+        if (values != null) {
+            for (UUID uuid : values) {
+                bsonValues.add(new BsonBinary(uuid));
+            }
+        }
+        return in(notNull("path", path), notNull("value", bsonValues));
```

Comment:
`bsonValues` is guaranteed to be not `null`, we should not check it with `notNull("value", bsonValues)`.